### PR TITLE
Add test coverage for ansible-playbook job with nonexisting_module

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Missing test coverage for ansible-playbook job with nonexisting_module, which covers BZ:2107577 and BZ:2028112

### Solution
Add test coverage for ansible-playbook job with nonexisting_module, which covers BZ:2107577 and BZ:2028112

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py::TestAnsibleREX -k 'test_positive_ansible_job_timeout_to_kill or test_positive_ansible_job_with_nonexisting_module'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8132
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_ansible.py::TestAnsibleREX -k test_positive_ansible_job_timeout_to_kill or test_positive_ansible_job_with_nonexisting_module --external-logging
Test Result : ==== 1 failed, 1 passed, 11 deselected, 169 warnings in 1751.88s (0:29:11) =====
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py::TestAnsibleREX -k 'test_positive_ansible_job_timeout_to_kill or test_positive_ansible_job_with_nonexisting_module'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8138
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_ansible.py::TestAnsibleREX -k test_positive_ansible_job_timeout_to_kill or test_positive_ansible_job_with_nonexisting_module --external-logging
Test Result : ========= 2 passed, 11 deselected, 132 warnings in 1281.33s (0:21:21) ==========
```

File: tests/foreman/api/test_ansible.py [hunk 1]
```diff
@@ -535,8 +535,8 @@ def test_positive_ansible_localhost_job_on_host(
         assert [i['output'] for i in result if i['output'] == 'Exit status: 0']
 
     @pytest.mark.no_containers
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_ansible.py:537]: It is required as we've a limitation of running a JobInvocation on actual VM host instead of container host

**Review by amolpati30 — APPROVED** [review]

File: tests/foreman/api/test_ansible.py [hunk 2]
```diff
@@ -690,3 +690,70 @@ def test_positive_ansible_job_privilege_escalation(
...
+
+        :steps:
+            1. Register a content host with satellite
+            2. Run Ansible playbook with nonexisting_module
+            3. Verify playbook fails and the Ansible REX job fails on Satellite
+
+        :expectedresults: Satellite job fails with the error and non-zero exit status,
+            when using a playbook with the nonexisting_module module.
+
+        :BZ: 2107577, 2028112
```
Comment (shubhamsg199) [inline tests/foreman/api/test_ansible.py:712]: Can you also add jira tracker cards here

**Review by shubhamsg199 — APPROVED** [review]
Ack, one minor comment

**Pull request merged**